### PR TITLE
211 restructure configuration format

### LINKED ISSUE #211: Restructure configuration format
Restructure the configuration so that there are no user-defined keys

Change these
- grids
- profiles
- dose rates

Add required/optional keys like this
https://github.com/NCAR/tuv-x/blob/f1c699c1e2a0f6355018767471aa4d3ed5382703/src/core.F90#L89-L99

Update docs where applicable

### PR DESCRIPTION
**Pull request by K20shores**
closes #211 

[New documentation for this branch](https://ncar.github.io/tuv-x/branch/211-restructure-configuration-format/)

### CODE REVIEW

File: src/profile_warehouse.F90 [hunk 1]
```diff
@@ -94,39 +94,34 @@ function constructor( config, grid_warehouse ) result( profile_warehouse )
...
                                              profile_obj%val_%units( ) ), &
                       "Profile '"//profile_obj%val_%handle_//             &
                       "' duplicated in profile warehouse." )
-      temp_profiles = profile_warehouse%profiles_
-      deallocate( profile_warehouse%profiles_ )
-      allocate( profile_warehouse%profiles_( size( temp_profiles ) + 1 ) )
-      profile_warehouse%profiles_( 1 : size( temp_profiles ) ) =              &
-          temp_profiles(:)
-      profile_warehouse%profiles_( size( temp_profiles ) + 1 ) = profile_obj
-      deallocate( temp_profiles )
```
Comment (K20shores) [inline src/profile_warehouse.F90:128]: @mattldawson I don't know if there was a reason we had for making the profiles and copying the pointers as the array grew, but I removed. I did have to add line 249 below in the `profile_warehouse%exists` function, not sure if that's appropriate or not

File: src/profile_warehouse.F90 [hunk 2]
```diff
@@ -94,39 +94,34 @@ function constructor( config, grid_warehouse ) result( profile_warehouse )
     ! local variables
     type(config_t)                 :: profile_config
     class(iterator_t), pointer     :: iter
-    type(profile_ptr), allocatable :: temp_profiles(:)
     type(profile_ptr)              :: profile_obj
     character(len=32)              :: keychar
     type(string_t)                 :: aswkey
     character(len=*), parameter    :: Iam = "profile warehouse constructor: "
+    integer                        :: profile_idx = 1
```
Comment (mattldawson) [inline src/profile_warehouse.F90:101]: this is another really strange fortran thing. If you define a variable when you declare it, fortran makes it static. I have no idea why. I would just remove the `= 1` here and set it to 1 before you enter the loop.

File: src/profile_warehouse.F90 (hunk 1 above)
Comment (mattldawson) [reply src/profile_warehouse.F90:128]: yeah, I like your way better.

File: src/profiles/from_csv_file.F90 [hunk 3]
```diff
@@ -57,15 +57,15 @@ function constructor( config, grid_warehouse ) result( this )
     real(dk), allocatable :: zdata(:)
     real(dk), allocatable :: profile(:)
     class(interpolator_t), pointer :: theInterpolator
-    type(string_t) :: required_keys(4), optional_keys(3)
+    type(string_t) :: required_keys(5), optional_keys(4)
```
Comment (mattldawson) [inline src/profiles/from_csv_file.F90:60]:
```suggestion
    type(string_t) :: required_keys(5), optional_keys(2)
```

File: docs/source/user_guide.rst [hunk 4]
```diff
@@ -160,8 +163,8 @@ For stand-alone TUV-x, you will also need to provide a ``time`` grid
 (units: ``hours``) that provides the time of day for which to calculate
 the photolysis and/or dose rates.
 
-There are two key-value pairs that are required for every grid, ``type``
-and ``units``.
+There are three key-value pairs that are required for every grid, ``type``
+, ``units``, and name.
```
Comment (mattldawson) [inline docs/source/user_guide.rst:167]:
```suggestion
, ``units``, and ``name``.
```

**Review by mattldawson — CHANGES_REQUESTED** [review]
looks good! just a very minor doc change, and one array size change

**Review by mattldawson — APPROVED** [review]

**Pull request merged**